### PR TITLE
util: support for mariadb-java-client 3.0.4

### PR DESCRIPTION
**Pull request by pierre**
See https://github.com/killbill/killbill-oss-parent/pull/366 and https://github.com/killbill/killbill-commons/pull/116.

### CODE REVIEW

**Comment by pierre** [comment]
Note: e2e test `KillBillIntegrationTests::TestWithDatesAndTimezones` is failing as expected.

File: util/src/main/java/org/killbill/billing/util/export/dao/DatabaseExportDao.java [hunk 1]
```diff
@@ -94,21 +95,32 @@ public void exportDataForAccount(final DatabaseExportOutputStream out, final Int
         }
 
         final List<ColumnInfo> columnsForTable = new ArrayList<ColumnInfo>();
+        // Separate lookup table, to keep the ordering of the columns
+        final Map<String, Integer> columnsLookup = new TreeMap<>(String.CASE_INSENSITIVE_ORDER); // Ignore casing (for H2)
```
Comment (sbrossie) [inline util/src/main/java/org/killbill/billing/util/export/dao/DatabaseExportDao.java:99]: Why do we need to sort the keys -> isn't the point of the value (the index) to specify which column comes first?

File: util/src/test/java/org/killbill/billing/util/export/dao/TestDatabaseExportDao.java [hunk 2]
```diff
@@ -86,7 +86,7 @@ public Void withHandle(final Handle handle) throws Exception {
         final String newDump = getDump();
 
         Assert.assertEquals(newDump, "-- accounts record_id|id|external_key|email|name|first_name_length|currency|billing_cycle_day_local|parent_account_id|is_payment_delegated_to_parent|payment_method_id|reference_time|time_zone|locale|address1|address2|company_name|city|state_or_province|country|postal_code|phone|notes|migrated|created_date|created_by|updated_date|updated_by|tenant_record_id\n" +
-                                     String.format("%s|%s|%s|%s|%s|%s||||false||%s|%s|||||||||||false|%s|%s|%s|%s|%s", internalCallContext.getAccountRecordId(), accountId, accountId, accountEmail, accountName, firstNameLength, "1970-05-24T18:33:02.000+00:00", timeZone,
+                                     String.format("%s|%s|%s|%s|%s|%s||||true||%s|%s|||||||||||false|%s|%s|%s|%s|%s", internalCallContext.getAccountRecordId(), accountId, accountId, accountEmail, accountName, firstNameLength, "1970-05-24T18:33:02.000+00:00", timeZone,
```
Comment (sbrossie) [inline util/src/test/java/org/killbill/billing/util/export/dao/TestDatabaseExportDao.java:89]: 🙄

File: util/src/main/java/org/killbill/billing/util/export/dao/DatabaseExportDao.java (hunk 1 above)
Comment (pierre) [reply util/src/main/java/org/killbill/billing/util/export/dao/DatabaseExportDao.java:99]: We actually don't need the ordering, we just need a map with case-insensitive keys.

Comment (sbrossie) [reply util/src/main/java/org/killbill/billing/util/export/dao/DatabaseExportDao.java:99]: Right, so technically you don't need a `TreeMap` (but not that it matters)

**Pull request merged**